### DISCUSSION TITLE
What versions of searx or(another search engine) have snippets/autoanswers exactly like google?

### DISCUSSION THREAD

**Discussion by Username23453** [searxng/searxng#2065] [General]
and also show the date of each article in the search results? because right now searx doesnt have either. 
-Thanks

**Comment by ahmad-alkadri** [comment]
I've been using searxng installed locally on docker for a while. By snippets, do you mean something like this:

<img width="1474" alt="Screenshot 2023-01-03 at 08 37 57" src="https://user-images.githubusercontent.com/22837764/210316231-23c7c318-8cc1-49d7-b82a-dd14e191a0ba.png">

If that's what you want, it does appear there from time to time, though probably not always. Moreover, about the date and time of each article, searxng also has it:

<img width="1425" alt="Screenshot 2023-01-03 at 08 33 21" src="https://user-images.githubusercontent.com/22837764/210316406-1b4e6a41-b9fa-4612-b038-d280e6507702.png">

**Reply by Username23453** [reply]
Thanks for your reply. Try "is vanilla antibacterial?" and see if it provides snippets/autoanswers, like google would. I know "searx.thegpm.org" does not. But searx.puffyan.us does provide snippets. But searx.puffyan.us gets errors with duckduckgo & Brave search engine it uses. -Thanks!

**Comment by return42** [comment]
FWIW:

- searx.puffyan.us missed maintenance / not updated since a longer time 2022.09.09-eb3d185e 
- searx.thegpm.org is not SearXNG .. can't speack for searx instance, but I see it an old version

.. look out for more updated instances at https://searx.space/

> Try "is vanilla antibacterial?" and see if it provides snippets/autoanswers, 

Try it by yourself [`is vanilla antibacterial :en`](https://darmarit.org/searx/search?q=is+vanilla+antibacterial+%3Aen) .. you need to select the right languages .. AFAIK most of the "answers" from ddg & google are 'en'.

**Reply by Username23453** [reply]
@return42, Thank you, thank you, 🙏🙏.    Very helpful, But i have no clue what you mean by this: "is vanilla antibacterial :en"(https://darmarit.org/searx/search?q=is+vanilla+antibacterial+%3Aen).  are you merely recommending that i always type ":en" in my searches?
-Thanks

**Reply by ahmad-alkadri** [reply]
> recommending that i always type ":en" in my searches? -Thanks

@Intellectual234 I don't think you have to. But many answers/snippets, especially from DDG & Google, are available in English language, thus if you want those you could try adding "**:en**" on your query.

**Reply by return42** [reply]
> try adding ":en" on your query

.. or select the language from the language menue

**Comment by Fmstrat** [comment]
Bringing this discussion back for a more specific question vs starting a new one. Is it possible to enable the "answer" areas of Google that aren't the text answers like above? For instance:

![image](https://github.com/user-attachments/assets/8d8f3617-12f4-4fab-8c3e-69a8da74be44)

or

![image](https://github.com/user-attachments/assets/08bc28f5-cfc9-44e5-9ea6-6b4ebf8399ad)

**Comment by return42** [comment]
> Is it possible to enable the "answer" areas of Google that aren't the text answers like above?

No :-o at the moment our google selectors only find the text answers shown above ... I haven't looked into it yet, but maybe we could customize our selectors to also find the example you showed ... would you like to open a feature request, maybe someone will have the time to take a closer look / thanks!

**Reply by Fmstrat** [reply]
Just put a PR with some screenshots up at https://github.com/searxng/searxng/pull/3734, not sure what the review process is, though.

Edit: Oops, wrong link ;)

**Comment by Fmstrat** [comment]
@return42 Took a look at the engine, looks like Google encodes some of the results in JS to be spit out into the DOM. I've made a little parser that pulls out items that contain the class currently being used to pull answers and it now catches the "what is the capital" item. Going to look at the time response, too. I'll put a draft PR together to confirm this is the way maintainers would want it done soon.
